libs/components/storybook/src/lib/icon-preview/icon-preview.service.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  DOCUMENT,
+  EnvironmentProviders,
+  Injectable,
+  computed,
+  effect,
+  inject,
+  provideAppInitializer,
+  resource,
+  signal,
+} from '@angular/core';
+import { SkyIconSvgResolverService } from '@skyux/icon';
+
+/* istanbul ignore next */
+export function provideIconPreview(): EnvironmentProviders[] {
+  return [
+    provideAppInitializer(() => {
+      inject(IconPreviewService);
+    }),
+  ];
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class IconPreviewService {
+  readonly #doc = inject(DOCUMENT);
+  readonly #previewLink = this.#doc.querySelector<HTMLLinkElement>(
```

Comment:
Looks for a `<link class="skyux-icons-preview"...>` and uses that to swap out the sprite.